extensions/VisualEditor/wikia/modules/ve/dm/ve.dm.WikiaTransclusionModel.js

```diff
@@ -2,178 +2,180 @@
+		ve.dm.WikiaTransclusionModel.super.call( this );
+	};
 
-/* Inheritance */
+	/* Inheritance */
 
-OO.inheritClass( ve.dm.WikiaTransclusionModel, ve.dm.MWTransclusionModel );
+	OO.inheritClass( ve.dm.WikiaTransclusionModel, ve.dm.MWTransclusionModel );
 
-/* Methods */
+	/* Methods */
 
-/**
- * @inheritdoc
- */
-ve.dm.WikiaTransclusionModel.prototype.fetchRequest = function ( titles, specs, queue ) {
-	//if we want to make another request, we should update the requests array as well
-	this.requests.push( this.fetchInfoboxParamsRequest( titles, specs, queue ) );
-
-	return ve.init.target.constructor.static.apiRequest( {
-		action: 'templateparameters',
-		titles: titles.join( '|' )
-	} )
-	.done( this.fetchRequestDone.bind( this, specs ) )
-	.always( this.fetchRequestAlways.bind( this, queue ) );
-};
-
-ve.dm.WikiaTransclusionModel.prototype.fetchInfoboxParamsRequest = function ( titles, specs, queue ) {
-	return ve.init.target.constructor.static.apiRequest( {
-		action: 'query',
-		prop: 'infobox',
-		titles: titles.join( '|' )
-	} )
-	.done( this.fetchInfoboxParamsRequestDone.bind( this, specs ) )
-	.always( this.fetchRequestAlways.bind( this, queue ) );
-};
-
-/**
- * @inheritdoc
- */
-ve.dm.WikiaTransclusionModel.prototype.fetchRequestDone = function ( specs, data ) {
-	var page, i, id;
-	if ( data && data.pages ) {
-		for ( id in data.pages ) {
-			page = data.pages[id];
-			if ( !specs[page.title] ) {
-				specs[page.title] = {
-					title: data.pages[id].title,
-					description: '',
-					params: {},
-					paramOrder: page.params
-				};
-			}
-			// Map parameters from flat array to collection where parameter name is the key and value is empty
-			// because later it's extended with ve.dm.MWTemplateSpecModel.prototype.getDefaultParameterSpec
-			for ( i = 0; i < page.params.length; i++ ) {
-				specs[page.title].params[page.params[i]] = {};
+	/**
+	 * @inheritdoc
+	 */
+	ve.dm.WikiaTransclusionModel.prototype.fetchRequest = function ( titles, specs, queue ) {
+		//if we want to make another request, we should update the requests array as well
+		this.requests.push( this.fetchInfoboxParamsRequest( titles, specs, queue ) );
+
+		return ve.init.target.constructor.static.apiRequest( {
+			action: 'templateparameters',
+			titles: titles.join( '|' )
+		} )
+		.done( this.fetchRequestDone.bind( this, specs ) )
+		.always( this.fetchRequestAlways.bind( this, queue ) );
+	};
+
+	ve.dm.WikiaTransclusionModel.prototype.fetchInfoboxParamsRequest = function ( titles, specs, queue ) {
+		return ve.init.target.constructor.static.apiRequest( {
+			action: 'query',
+			prop: 'infobox',
+			titles: titles.join( '|' )
+		} )
+		.done( this.fetchInfoboxParamsRequestDone.bind( this, specs ) )
+		.always( this.fetchRequestAlways.bind( this, queue ) );
+	};
+
+	/**
+	 * @inheritdoc
+	 */
+	ve.dm.WikiaTransclusionModel.prototype.fetchRequestDone = function ( specs, data ) {
+		var page, i, id;
+		if ( data && data.pages ) {
+			for ( id in data.pages ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)